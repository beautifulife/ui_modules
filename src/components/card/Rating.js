import React, { Fragment } from 'react';
import './Rating.scss';

const Rating = ({ rating, ratingDesc, type }) => {
  const renderRateCircle = () => {
    const renderedCircles = [];

    for (let i = 1; i <= 5; i++) {
      const isFilled = i <= rating;

      renderedCircles.push(
        <span key={i} className={`circle ${isFilled ? 'filled' : ''}`} />
      );
    }

    return renderedCircles;
  };

  return (
    <Fragment>
      {type === 'vertical' ? (
        <div className="Rating">
          <div className="Rating__wrapper">
            <div className="Rating__rate">{renderRateCircle()}</div>
            <div className="Rating__desc">{ratingDesc}</div>
          </div>
        </div>
      ) : (
        <div className="Rating vertical">
          <div className="Rating__desc">{ratingDesc}</div>
          <div className="Rating__desc">{ratingDesc}</div>
          <div className="Rating__rate">{renderRateCircle()}</div>
        </div>
      )}
    </Fragment>
  );
};

export default Rating;
