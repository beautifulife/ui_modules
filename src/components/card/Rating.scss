.Rating {
  &.vertical {
    height: 55px;
    padding: 0 10px;
    border-top: 1px solid #eee;

    &::before {
      content: '';
      display: inline-block;
      height: 100%;
      vertical-align: middle;
    }
  }

  &__wrapper {
    display: inline-block;
    width: 100%;
    vertical-align: middle;
  }

  &.horizontal {
    display: block;
    width: 200px;
    padding: 10px;
  }

  &__rate {
    padding: 5px 0;
    line-height: 0.8rem;

    .circle {
      display: inline-block;
      width: 1rem;
      height: 1rem;
      margin-right: 3px;
      border-radius: 50%;
      background-color: #d4d4d4;
      text-indent: -9999px;

      &.filled {
        background-color: #ffd700;
      }
    }

    .author {
      display: inline-block;
      margin-left: 7px;
      padding-left: 7px;
      border-left: 1px solid #d4d4d4;
      color: rgba(0, 0, 0, 0.7);
      line-height: 16px;
      vertical-align: top;
    }
  }

  &__desc {
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    color: rgba(0, 0, 0, 0.7);

    &.title {
      color: #000000;
      font-weight: bold;
      margin-bottom: 15px;
    }

    &.long {
      margin-bottom: 20px;
      overflow: visible;
      white-space: pre-wrap;
    }
  }
}
