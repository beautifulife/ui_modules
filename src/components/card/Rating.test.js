import React from 'react';
import { shallow } from 'enzyme';
import Rating from './Rating';

describe('Rating', () => {
  describe('about render', () => {
    const wrapper = shallow(<Rating />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('render horizontal when cardType is horizontal', () => {
      wrapper.setProps({ type: 'horizontal' });
      expect(wrapper.find('.Rating.horizontal').length).toBe(1);
    });
  });
});
