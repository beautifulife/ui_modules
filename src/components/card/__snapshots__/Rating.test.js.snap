// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rating
    author=""
    ratingDesc=""
    type="vertical"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="Rating vertical"
        >
          <div
            className="Rating__wrapper"
          >
            <div
              className="Rating__rate"
            >
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
            </div>
            <div
              className="Rating__desc"
            >
              
            </div>
          </div>
        </div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="Rating__wrapper"
          >
            <div
              className="Rating__rate"
            >
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
              <span
                className="circle "
              >
                평가 총 5점 중 undefined점
              </span>
            </div>
            <div
              className="Rating__desc"
            >
              
            </div>
          </div>,
          "className": "Rating vertical",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Rating__rate"
              >
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
              </div>,
              <div
                className="Rating__desc"
              >
                
              </div>,
            ],
            "className": "Rating__wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>,
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>,
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>,
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>,
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>,
                ],
                "className": "Rating__rate",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "평가 총 5점 중 undefined점",
                    "className": "circle ",
                  },
                  "ref": null,
                  "rendered": "평가 총 5점 중 undefined점",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "평가 총 5점 중 undefined점",
                    "className": "circle ",
                  },
                  "ref": null,
                  "rendered": "평가 총 5점 중 undefined점",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": "평가 총 5점 중 undefined점",
                    "className": "circle ",
                  },
                  "ref": null,
                  "rendered": "평가 총 5점 중 undefined점",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "host",
                  "props": Object {
                    "children": "평가 총 5점 중 undefined점",
                    "className": "circle ",
                  },
                  "ref": null,
                  "rendered": "평가 총 5점 중 undefined점",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "host",
                  "props": Object {
                    "children": "평가 총 5점 중 undefined점",
                    "className": "circle ",
                  },
                  "ref": null,
                  "rendered": "평가 총 5점 중 undefined점",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "Rating__desc",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="Rating vertical"
          >
            <div
              className="Rating__wrapper"
            >
              <div
                className="Rating__rate"
              >
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
              </div>
              <div
                className="Rating__desc"
              >
                
              </div>
            </div>
          </div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="Rating__wrapper"
            >
              <div
                className="Rating__rate"
              >
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
                <span
                  className="circle "
                >
                  평가 총 5점 중 undefined점
                </span>
              </div>
              <div
                className="Rating__desc"
              >
                
              </div>
            </div>,
            "className": "Rating vertical",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="Rating__rate"
                >
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>
                  <span
                    className="circle "
                  >
                    평가 총 5점 중 undefined점
                  </span>
                </div>,
                <div
                  className="Rating__desc"
                >
                  
                </div>,
              ],
              "className": "Rating__wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="circle "
                    >
                      평가 총 5점 중 undefined점
                    </span>,
                    <span
                      className="circle "
                    >
                      평가 총 5점 중 undefined점
                    </span>,
                    <span
                      className="circle "
                    >
                      평가 총 5점 중 undefined점
                    </span>,
                    <span
                      className="circle "
                    >
                      평가 총 5점 중 undefined점
                    </span>,
                    <span
                      className="circle "
                    >
                      평가 총 5점 중 undefined점
                    </span>,
                  ],
                  "className": "Rating__rate",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "host",
                    "props": Object {
                      "children": "평가 총 5점 중 undefined점",
                      "className": "circle ",
                    },
                    "ref": null,
                    "rendered": "평가 총 5점 중 undefined점",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": "2",
                    "nodeType": "host",
                    "props": Object {
                      "children": "평가 총 5점 중 undefined점",
                      "className": "circle ",
                    },
                    "ref": null,
                    "rendered": "평가 총 5점 중 undefined점",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": "3",
                    "nodeType": "host",
                    "props": Object {
                      "children": "평가 총 5점 중 undefined점",
                      "className": "circle ",
                    },
                    "ref": null,
                    "rendered": "평가 총 5점 중 undefined점",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": "4",
                    "nodeType": "host",
                    "props": Object {
                      "children": "평가 총 5점 중 undefined점",
                      "className": "circle ",
                    },
                    "ref": null,
                    "rendered": "평가 총 5점 중 undefined점",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": "5",
                    "nodeType": "host",
                    "props": Object {
                      "children": "평가 총 5점 중 undefined점",
                      "className": "circle ",
                    },
                    "ref": null,
                    "rendered": "평가 총 5점 중 undefined점",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "Rating__desc",
                },
                "ref": null,
                "rendered": "",
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
